### DISCUSSION TITLE
Disable password generator ?

### DISCUSSION THREAD

**Discussion by Alice-Deane** [pglombardo/PasswordPusher#1063] [Q&A]
Hello it is me again ... 

I would like to know if there is any ways to disable the password generator option ? 
I can't find any ENV variable to do this, I may I miss something.

Thanks again ! 
Have a good day :)

**Comment by pglombardo** [comment]
Hi @Alice-Deane - welcome back!  There is no way currently to disable the password generator.  Can I ask why you would want to?

**Comment by Alice-Deane** [comment]
Hi @pglombardo, 
I would like to disable it because we don't really have a use for it. We use another system to generate our passwords.

**Reply by pglombardo** [reply]
Ok makes sense.  I created #1067.  I'll add an option to disable the password generator soon.

**Comment by pglombardo** [comment]
Hi @Alice-Deane - just an update for this request.  I've added this request as an option for all users on pwpush.com.   This will make it into the open source eventually although I don't have any ETA as of yet.

![Screenshot 2024-09-01 at 16 48 37](https://github.com/user-attachments/assets/2db783fa-e830-42c7-9cd4-50ca78a33724)

**Comment by EarMaster** [comment]
Is there an update on this for self-hosted? Users have reported repeatedly that they accidentally clicked the "generate password" button because it is located like a "submit" button right below the secret field and then it replaced the actual secret.

**Reply by pglombardo** [reply]
Hi @EarMaster -We haven't gotten to configuring the visibility of the Password Generator.  This is partly because we plan to replace it entirely.

In the mean time, we'll add a dialog to confirm before replacing the content: https://github.com/pglombardo/PasswordPusher/issues/3585
